### DISCUSSION TITLE
Merch

### DISCUSSION THREAD

**Discussion by simonv3** [funmusicplace/mirlo#224] [Ideas]
Mirlo should figure out a way (whether natively, through a plug-in system with artisan.coop or shopify, or whatever) allow artists to sell merchandise of any type. 

Edit, from LLK what bandcamp does:

1. it creates an "order" row in a board with shipping informations
2. Then once you've shipped the item you can send a message to that person, it goes, you click on "mark as shipped" a pop up appears and you can enter a tracking number and the url of the tracking service, plus a small personalized message

**Comment by LouisLouiseKay** [comment]
This is one of the features that would imediately set us apart from the crowd of digital only marketplaces.

**Comment by bentooth** [comment]
I believe this would widen the potential user base outside the power user seeking digital downloads. We can integrate something like Shopify so artists can use their dashboards and analytics to manage their products...but the pricing starts at $40/mo 👎 

 Or we can roll our own  👀  🏗️

**Comment by Cwpute** [comment]
I'd advise against a native solution right away, it's quite out of scope for now. Maybe another coop could be created to handle this, but until now i think all of this should be handled by someone else than Mirlo.

Until then, if you allowed artist to show a link on their page specifically for merchandising, it would certainly help and serve as a working solution for very little work in return :)

**Reply by simonv3** [reply]
This has been discussed to integrate a bit more with [https://artisans.coop/](https://artisans.coop)

**Reply by Cwpute** [reply]
Oh great ! that sounds wonderful.

**Comment by simonv3** [comment]
# Step 1
 
* [x] Allow artists to link to a merch store. This should be a unique link that if it exists will be a new tab and will redirect the artist. Add it to the `artist.merchLink` column. 

# Step 2

Fleshing out a potential database schema:

```
model Merch {
  id               String     @id @default(uuid()) @db.Uuid
  title 
  description
  minPrice
  allowVariablePrice
  quantityRemaining
  beginsShipping
  sku
  includePurchaseTrackGroupId
  isPublic
  itemType // One of ['Compact Disk (CD)', 'Cassette', etc]
  createdAt
  updatedAt
  deletedAt
}

model MerchImage {
  id 
  url String[]
}

model MerchShipping {
  homeCountry    String
  destination   String
  costUnit
  currency
  costExtraUnit
  merchId
}

model MerchOptionType { 
  optionName // eg. "shirt size"
  merchId
}

model MerchOption {
  optionTypeId 
  name // eg. "small"
  quantity 
  sku
}

model MerchPurchase {
  merchId
  currencyPaid
  amountPaid
  stripeTransactionKey
}

```

# Step 3: implement a merch form

* [x] Basically the above options but in form shape

# Step 4: Display merch

* [x] If there is a `artist.merchLink` and merch associated to an artist the tab should show both the link and the existing merch.
* [x] Merch that has 0 quantity should be marked as sold out.

# Step 4: Merch Purchase

* [x] Allow users to purchase merch

**Comment by Cwpute** [comment]
If you're going to have an item type list the artist will choose from, also allow a catch-all "Other" type of item, with the ability to type what item type it is. Optionally, integrate those typed-in types into your datasets so as to keep tabs on trending new merch, merch with slight-differences that matters, etc…

Le 28 juin 2024 20:26:37 GMT+02:00, Psi ***@***.***> a écrit :

**Comment by tshiunghan** [comment]
I could imagine this feature as parked under Fun Music Place as a project, different from Mirlo. 

Whatever we decide, we should try to inform the user about any surcharges as early as possible in the process. 

Source: https://www.nngroup.com/articles/ecommerce-taxes-fees/

**Comment by simonv3** [comment]
This is implemented though it still needs to be properly taken for a test ride. But that's what bug issues are for.
